packages/core-data/src/queried-data/reducer.js

```diff
@@ -110,7 +110,8 @@ export function items( state = {}, action ) {
 				[ context ]: {
 					...state[ context ],
 					...action.items.reduce( ( accumulator, value ) => {
-						const itemId = value[ key ];
+						const itemId = value?.[ key ];
```

Comment:
It turns out that `undefined` is a "valid" item ID, and site settings are keyed using it.

## Screenshot
![CleanShot 2024-06-06 at 00 13 33](https://github.com/WordPress/gutenberg/assets/240569/3e735532-62db-4a4c-bae0-5697985909f1)